qiskit/util.py

```diff
@@ -163,3 +163,40 @@ def is_main_process():
             and (sys.version_info[1] == 5 or sys.version_info[1] == 6)
             and mp.current_process().name != 'MainProcess')
     )
+
+
+def apply_prefix(value: float, unit: str) -> float:
```

Comment:
I just leave the `apply_prefix` function as is (it's just moved from providers/models/backendproperties.py). The case of `unit='s'` is cared within [`DelayInDt` pass](https://github.com/Qiskit/qiskit-terra/blob/db1b243c8e90eba77748544c6427dc22fc3ed096/qiskit/transpiler/passes/scheduling/delayindt.py#L62).

Reply:
Can you please update `backend_properties` to make use of this function please?

Reply:
Why do you think it should be handled there rather than here?

Reply:
This PR introduces `circuit.Delay` instruction which allows to specify unit so that it requires unit conversion as well as `BackendProperties`. @ajavadia suggested to use a common logic for that (to avoid code duplication). The original conversion code was in `BackendProperties._apply_suffix`.
I think this is a good timing to make it public (and place it more common path) because it is now called by two separated functions. I've already updated `BackendProperties._apply_suffix` to call this as you expected (pleas see 686350a).